### ISSUE TITLE
Is there a way to export the dash layout as an html?

### ISSUE THREAD

**Issue by tianjixuetu** [plotly/dash#1056]
[I find this question have asked in 2017,but,I want to ask again.](https://github.com/plotly/dash/issues/145)

dash have a function to save the layout as a html?

I am a quant, and do many simulates,and I use dash to plot the results in the browser,but,I find,I have to caculate it again or save the results from the browser by myself , in order to check the results again.

so is there a convenient way to save the dash layout?

**Comment by exzhawk** [comment]
Almost another two years later, still no official solution. I love using Dash and sometimes I need to share my charts with people who don't know Python. So I made this workaround.
https://gist.github.com/exzhawk/33e5dcfc8859e3b6ff4e5269b1ba0ba4

To see the demo:
Download the Python file and run it. You will see a simple bar chart as well as a table. After clicking the left upper button, you will find a folder besides the Python file named "target". In which you will have a fully offline static HTML file with all other files needed (hopefully). You may stop Python and open the `index.html` directly inside the "target" folder. You may send the whole "target" folder to others and let them open it without any Python runtime or HTTP server, just with a browser.

Under the hood:
The code is actually simple but hacky. The dash web page relies on ajax communication to fetch JSON data to render. So the `make_static` function downloads all resources and the JSON, patches the JSON into the `index.html` file, and tells the scripts in the page to get data from `index.html` instead of requesting to Python backend.

Limitation:
It's a static HTML file bro. No fancy callbacks anymore. You can only save the initial state rather than the ideal "current layout when the save button is clicked",

**Comment by monsieurhannes** [comment]
Yes, that feature would be great. I like @exzhawk approach and personally it would be enough for me to just export the _current state_ as html, but the _initial state_ is not very useful for my project as I integrated `dcc.Upload` to get the data in the first place.

**Comment by AhmedThahir** [comment]
Any updates on this?
I really want `current_state` as html

**Comment by armingerami** [comment]
Thanks for sharing your solution!
